Mathlib/Order/OrdinalApproximantsFixedPoints.lean

```diff
@@ -0,0 +1,365 @@
+
+## Main theorems
+* `OrdinalApprox.lfp_is_lfp_approx`: The approximation of
+  the least fixed point eventually reaches the least fixed point
+* `OrdinalApprox.gfp_is_gfp_approx`: The approximation of
+  the greatest fixed point eventually reaches the greatest fixed point
+
+## Tags
+
+fixed point, complete lattice, monotone function, ordinals, approximation
+-/
+
+namespace Cardinal
+
+universe u
+variable {α : Type u}
+variable (g: Ordinal → α)
+
+open Cardinal Ordinal SuccOrder Function
+
+/-- Limiation of an ordinal defined function to an initial segment -/
+def limitation : { i : Ordinal // i < (ord $ succ #α)} → α := λ i => g i
+
+theorem limitation_def : ∀ i, limitation g i = g i := by
+  intro h; unfold limitation; exact rfl
+
+lemma mk_initialSeg_Subtype (ℵ : Cardinal ): #{ i : Ordinal // i < (ord  ℵ)} = lift.{u+1, u} ℵ := by
+  simpa using mk_initialSeg (ord ℵ)
+
+theorem limitation_not_injective: ¬ Injective (limitation g) := by
+  intro h_inj
+  have h₁ := by apply lift_mk_le_lift_mk_of_injective h_inj;
+  rw[mk_initialSeg_Subtype (succ #α), Cardinal.lift_lift, Cardinal.lift_le] at h₁
+  have h₂ := not_le_of_lt (Cardinal.succ_lt #α)
+  exact h₂ h₁
+
+end Cardinal
+
+
+
+namespace Monotone
+
+
+universe u v
+variable {α : Type u}
+variable {β : Type v}
+variable [Preorder α] [PartialOrder β] (f: α → β)
+
+/-- If a monotone function is equal at two points, it is equal between all of them -/
+theorem monotone_stabilizing {a₁ a₂ : α} (h_mon : Monotone f) (h_fa : f a₁ = f a₂):
+    ∀ i, a₂ ≥ i → i ≥ a₁ → f i = f a₁ := by
+  intro i h₂ h₁
+  apply le_antisymm
+  · rw[h_fa]; exact h_mon h₂
+  · exact h_mon h₁
+
+/-- If an antitone function is equal at two points, it is equal between all of them -/
+theorem antitone_stabilizing {a₁ a₂ : α} (h_anti : Antitone f) (h_fa : f a₁ = f a₂):
+    ∀ i, a₂ ≥ i → i ≥ a₁ → f i = f a₁ := by
+  intro i h₂ h₁
+  apply le_antisymm
+  · exact h_anti h₁
+  · rw[h_fa]; exact h_anti h₂
+
+end Monotone
+
+/-We loosly follow the proof from Frederico Enchenique in A Short and Constructive Proof of
+  Tarski's fixed-point theorem published in the International Journal of Game Theory Volume 33(2),
+  June 2005, pages 215-218. -/
+
+namespace OrdinalApprox
+
+universe u
+variable {α : Type u}
+variable [CompleteLattice α] (f : α →o α)
+
+open Function fixedPoints Cardinal Order OrderHom
+
+set_option linter.unusedVariables false in
+/-- Ordinal approximants of the least fixed points -/
+def lfp_approx (a : Ordinal.{u}) : α :=
+  sSup { f (lfp_approx b) | (b : Ordinal) (h : b < a) }
+termination_by lfp_approx a => a
+decreasing_by exact h
+
+theorem lfp_approx_monotone : Monotone (lfp_approx f) := by
+  unfold Monotone; intros a b h; unfold lfp_approx
+  refine sSup_le_sSup ?h; simp
+  intros a' h'
+  use a'; apply And.intro; exact lt_of_lt_of_le h' h
+  exact rfl
+
+/-- Ordinal approximants of the least fixed points as an order homomorphism -/
+def lfp_approx_hom : Ordinal.{u} →o α where
```

Comment:
```suggestion
def lfpApproxHom : Ordinal.{u} →o α where
```

Reply:
I removed this as it was not necessary for any of the theorems.